File: Algorithm/Merge Sorted Array.java
public class Solution {
    public void merge(int A[], int m, int B[], int n) {
        if (A == null || B == null) {
            return;
        }
        int tail = m + n - 1;
        m = m - 1;
        n = n - 1;
        while (m >= 0 && n >= 0) {
            if (A[m] >= B[n]) {
                A[tail--] = A[m--];
            } else {
                A[tail--] = B[n--];
            }
        }
        while (m >= 0) {
            A[tail--] = A[m--];
        }
        while (n >= 0) {
            A[tail--] = B[n--];
        }
    }
}

File: Algorithm/JavaScript/Merge Sorted Array.js
/**
 * @param {number[]} nums1
 * @param {number} m
 * @param {number[]} nums2
 * @param {number} n
 * @return {void} Do not return anything, modify nums1 in-place instead.
 */
var merge = function(nums1, m, nums2, n) {
    if (!nums1.length || !nums2.length) {
        return;
    }
    let insertPos = m + n - 1;
    m--;
    n--;
    while (n >= 0) {
        nums1[insertPos--] = nums1[m] >= nums2[n] ? nums1[m--] : nums2[n--];
    }
};

console.log(merge([1,2,3,0,0,0], 3, [2,5,6], 3));